nucypher/network/sensors.py

```diff
@@ -0,0 +1,186 @@
+        failure.raiseException()
+
+    def status(self) -> bool:
+        """Returns current indication of availability"""
+        return self.score > self.SENSOR_SENSITIVITY
+
+    @property
+    def running(self) -> bool:
+        return self.__task.running
+
+    def start(self, now: bool = False):
+        if not self.running:
+            self._start_time = maya.now()
+            d = self.__task.start(interval=self.FAST_INTERVAL, now=now)
+            d.addErrback(self.handle_measurement_errors)
+
+    def stop(self) -> None:
+        if self.running:
+            self.__task.stop()
+
+    def maintain(self) -> None:
+        self.log.debug(f"Measuring node availability current score is {self.score}.")
+        known_nodes_is_smaller_than_sample_size = len(self._ursula.known_nodes) < self.SAMPLE_SIZE
+
+        # If there are no known nodes or too few known nodes, skip this round...
+        # ... but not for longer than the maximum allotted alone time
+        if known_nodes_is_smaller_than_sample_size:
+            if not self._ursula.lonely and self.enforce_loneliness:
+                now = maya.now().epoch
+                delta = now - self._start_time.epoch
+                if delta >= self.MAXIMUM_ALONE_TIME:
+                    self.severe_warning()
+                    reason = self.Solitary if not self._ursula.known_nodes else self.Lonely
+                    self.shutdown_everything(reason=reason)
+            return
+
+        if self.__task.interval == self.FAST_INTERVAL:
+            now = maya.now().epoch
+            delta = now - self._start_time.epoch
+            if delta >= self.SEEDING_DURATION:
+                # Slow down
+                self.__task.interval = self.SLOW_INTERVAL
+                return
+
+        # All systems go
+        self.measure()
+        delta = maya.now() - self._start_time
+        self.log.info(f"Current availability score is {self.score} uptime: {delta}")
+        self.issue_warnings()
+
+    def issue_warnings(self, cascade: bool = True) -> None:
+        warnings = sorted(self.warnings.items(), key=lambda t: t[0])
+        for threshold, action in warnings:
+            if self.score <= threshold:
+                action()
+                if not cascade:
+                    # Exit after the first active warning is issued
+                    return
+
+    def sample(self, quantity: int) -> list:
+        population = tuple(self._ursula.known_nodes._nodes.values())
+        ursulas = random.sample(population=population, k=quantity)
+        return ursulas
+
+    @property
+    def score(self) -> float:
+        return self.__score
+
+    def record(self, result: bool = None) -> None:
+        """Score the result and cache it."""
+        if result is None:
+            return
+        score = int(result) + self.CHARGE_RATE * self.__score
+        if score >= self.MAXIMUM_SCORE:
+            return
```

Comment:
Can we just set score as MAXIMUM_SCORE or do you leave the original score (higher than MAXIMUM_SCORE) for the smoothing out of subsequent scores over time?

Reply:
:heavy_check_mark: